crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -6,27 +6,37 @@ use ruff_formatter::{format_args, write, FormatError, SourceCode};
 use ruff_python_ast::node::AnyNodeRef;
 use ruff_python_ast::prelude::AstNode;
 use ruff_text_size::{TextLen, TextRange, TextSize};
+use rustpython_parser::ast::Ranged;
 
 /// Formats the leading comments of a node.
-pub(crate) fn leading_comments<T>(node: &T) -> FormatLeadingComments
+pub(crate) fn leading_node_comments<T>(node: &T) -> FormatLeadingComments
```

Comment:
I renamed the function because I now needed a way to either format the leading comments of a node OR a specific list of comments.